bindings/python/iota_sdk/types/block/metadata.py

```diff
@@ -2,66 +2,82 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from __future__ import annotations
-from enum import Enum
+from enum import Enum, IntEnum
 from dataclasses import dataclass
-from typing import List, Optional, Union, Dict
-from dacite import from_dict
+from typing import Optional
 from iota_sdk.types.common import HexStr, json
-from iota_sdk.types.payload import TaggedDataPayload, TransactionPayload
-from iota_sdk.utils import Utils
 
 
 @json
 @dataclass
-class Block:
-    """Represent the object that nodes gossip around the network.
+class BlockMetadata:
+    """Represents the metadata of a block.
+    Response of GET /api/core/v3/blocks/{blockId}/metadata.
 
     Attributes:
-        protocol_version: The protocol version with which this block was issued.
-        strong_parents: Blocks that are strongly directly approved.
-        weak_parents: Blocks that are weakly directly approved.
-        shallow_like_parents: Blocks that are directly referenced to adjust opinion.
-        max_burned_mana: The amount of Mana the Account identified by the IssuerId is at most willing to burn for this block.
-        payload: The optional payload of this block.
+        block_state: The block state.
+        tx_state: The transaction state.
+        block_failure_reason: The block failure reason.
+        tx_failure_reason: The transaction failure reason.
     """
+    block_id: HexStr
+    # TODO: verify if really optional:
+    # https://github.com/iotaledger/tips-draft/pull/24/files#r1293426314
+    block_state: Optional[BlockState] = None
+    tx_state: Optional[TransactionState] = None
+    block_failure_reason: Optional[BlockFailureReason] = None
+    tx_failure_reason: Optional[TransactionFailureReason] = None
 
-    protocol_version: int
-    strong_parents: List[HexStr]
-    weak_parents: List[HexStr]
-    shallow_like_parents: List[HexStr]
-    max_burned_mana: str
-    payload: Optional[Union[TaggedDataPayload,
-                      TransactionPayload]] = None
 
-    @classmethod
-    def from_dict(cls, block_dict: Dict) -> Block:
-        """
-        The function `from_dict` takes a dictionary that contains the data needed to
-        create an instance of the `Block` class.
+class BlockState(Enum):
+    """Describes the state of a block.
+
+    Attributes:
+        Pending: Stored but not confirmed.
+        Confirmed: Confirmed with the first level of knowledge.
+        Finalized: Included and can no longer be reverted.
+        Rejected: Rejected by the node, and user should reissue payload if it contains one.
+        Failed: Not successfully issued due to failure reason.
```

Comment:
Put the number in parenthesis?

Reply:
Actually I've just seen these are supposed to serialise as string, is it correct to assign them an integer at all? I don't know python enough

Reply:
Usually not done for regular Enum types that don't require conversion to their int representaion. In that case we use IntEnum's.

Reply:
The thing is that you HAVE to assign something, Python doesnt allow for Rust-like enums without assignment.

Reply:
Sorry I'm not following. Why are we assigning them integers here?

Reply:
Oh ok, I answered before seeing your response. Well that sucks but ok.

Reply:
Only for syntactic reason